lib/auth-header.js

```diff
@@ -7,10 +7,15 @@ module.exports = ({ scheme, token, username, password }) => {
     return `Token ${replace(token, config)}`;
   }
   if (scheme === 'basic') {
-    const basicAuth = [
-      replace(username, config),
-      replace(password, config),
-    ].join(':');
+    let basicAuth;
+
+    if (token) {
+      basicAuth = replace(token, config);
+    } else {
+      basicAuth = [replace(username, config), replace(password, config)].join(
```

Comment:
The token part doesn't look right. Basic auth is supposed to be username and password.

https://en.wikipedia.org/wiki/Basic_access_authentication

> In the context of an HTTP transaction, basic access authentication is a method for an HTTP user agent (e.g. a web browser) to provide a user name and password when making a request. In basic HTTP authentication, a request contains a header field in the form of Authorization: Basic <credentials>, where credentials is the Base64 encoding of ID and password joined by a single colon :.

Reply:
Agree. But Azure reimplement authentication in its own way

Reply:
You might want to add a comment which SCMs use token